src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -417,55 +398,22 @@ public void dropChannel(DropChannelRequest request) {
         request.getFullyQualifiedTableName(),
         getName());
 
-    try {
-      Map<Object, Object> payload = new HashMap<>();
-      payload.put(
-          "request_id", this.flushService.getClientPrefix() + "_" + counter.getAndIncrement());
-      payload.put("channel", request.getChannelName());
-      payload.put("table", request.getTableName());
-      payload.put("database", request.getDBName());
-      payload.put("schema", request.getSchemaName());
-      payload.put("role", this.role);
-      payload.put("is_iceberg", isIcebergMode);
-      Long clientSequencer = null;
-      if (request instanceof DropChannelVersionRequest) {
-        clientSequencer = ((DropChannelVersionRequest) request).getClientSequencer();
-        if (clientSequencer != null) {
-          payload.put("client_sequencer", clientSequencer);
-        }
-      }
-
-      DropChannelResponse response =
-          executeWithRetries(
-              DropChannelResponse.class,
-              DROP_CHANNEL_ENDPOINT,
-              payload,
-              "drop channel",
-              STREAMING_DROP_CHANNEL,
-              httpClient,
-              requestBuilder);
-
-      // Check for Snowflake specific response code
-      if (response.getStatusCode() != RESPONSE_SUCCESS) {
-        logger.logDebug(
-            "Drop channel request failed, channel={}, table={}, client={}, message={}",
-            request.getChannelName(),
-            request.getFullyQualifiedTableName(),
-            getName(),
-            response.getMessage());
-        throw new SFException(ErrorCode.DROP_CHANNEL_FAILURE, response.getMessage());
-      }
+    DropChannelRequestInternal dropChannelRequest =
+        new DropChannelRequestInternal(
+            this.storageManager.getClientPrefix() + "_" + counter.getAndIncrement(),
+            this.role,
+            request,
```

Comment:
same here, lets pass all the request fields directly instead of passing the request object.